### ISSUE TITLE
Counting MACs in "Pruning by Masking" mode

### ISSUE THREAD

**Issue by RGTPdyn** [VainF/Torch-Pruning#459]
Hi,
Is there an easy way to count MACs of masked model? using utils.count_ops_and_params return numbers as unpruned model.
Thanks!

**Comment by AvrahamRaviv** [comment]
Hi
Here is my solution:
```
def measure_macs_masked_model(self, model):
        # Get per-layer MACs (and params) as dictionaries
        macs_dict = tp.utils.op_counter.count_ops_and_params(model, self.example_inputs, layer_wise=True)[2]

        total_adjusted_macs = 0.0
        total_macs = 0.0

        for name, module in model.named_modules():
            if module in macs_dict and isinstance(module, torch.nn.Conv2d):
                macs_layer = macs_dict[module]
                weight = module.weight
                out_channels, in_channels, _, _ = weight.shape

                # Count fully-zeroed input channels:
                zeros_input = (torch.sum(weight, dim=(0, 2, 3)) == 0).sum().item()
                # Count fully-zeroed output channels:
                zeros_output = (torch.sum(weight, dim=(1, 2, 3)) == 0).sum().item()

                active_in_ratio = (in_channels - zeros_input) / in_channels
                active_out_ratio = (out_channels - zeros_output) / out_channels

                ratio = active_in_ratio * active_out_ratio
                total_adjusted_macs += macs_layer * ratio
                total_macs += macs_layer
        return total_adjusted_macs, total_macs
```
